### PR TITLE
Allow personal settings via info.xml and doc links

### PR DESCRIPTION
**Pull request by nickvergessen**
Ref https://github.com/nextcloud/server/pull/8232

### CODE REVIEW

File: nextcloudappstore/api/v1/release/info.xsd [hunk 1]
```diff
@@ -357,6 +364,10 @@
                         maxOccurs="unbounded"/>
             <xs:element name="admin-section" type="php-class" minOccurs="0"
                         maxOccurs="unbounded"/>
+            <xs:element name="personal" type="php-class" minOccurs="0"
```
Comment (nickvergessen) [inline nextcloudappstore/api/v1/release/info.xsd]: Supported with 13+
https://github.com/nextcloud/server/pull/4890

File: nextcloudappstore/api/v1/release/info.xsd [hunk 2]
```diff
@@ -242,6 +242,13 @@
         </xs:restriction>
     </xs:simpleType>
 
+    <xs:simpleType name="doc-url">
+        <xs:restriction base="xs:anyURI">
+            <xs:pattern value="(https://.+|(admin|user|developer)-[a-z]+[\-a-z]*[a-z]+)"/>
```
Comment (nickvergessen) [inline nextcloudappstore/api/v1/release/info.xsd]: Supported with 7+ 🙈

Comment (BernhardPosselt) [reply nextcloudappstore/api/v1/release/info.xsd]: base="xs:anyURI" is now wrong, should be base="non-empty-string"

File: nextcloudappstore/api/v1/release/info.xsd (hunk 1 above)
Comment (BernhardPosselt) [reply nextcloudappstore/api/v1/release/info.xsd]: also updating the docs would be nice https://github.com/nextcloud/appstore/blob/master/docs/developer.rst#app-metadata

E.g. what interface to implement or which base class to extend and what it does

**Comment by BernhardPosselt** [comment]
I will take care of the tests and parsing. I suppose we just ignore non urls on the store?

**Comment by BernhardPosselt** [comment]
Your regex does not work btw, test fails ;)

**Comment by BernhardPosselt** [comment]
Is there a use case for having 

    <admin>user-something</admin>

File: nextcloudappstore/api/v1/release/info.xsd (hunk 2 above)
Comment (BernhardPosselt) [reply nextcloudappstore/api/v1/release/info.xsd]: **admin-** prefix should probably only work in \<admin> tags right?

Comment (nickvergessen) [reply nextcloudappstore/api/v1/release/info.xsd]: Well it's not enforced on server side, but yeah it's the case at the moment, so I guess we can do that.

**Comment by nickvergessen** [comment]
Btw, when you changed the info.xsd, please leave a note at https://github.com/nextcloud/server/pull/8232 or update the file directly.

**Comment by BernhardPosselt** [comment]
@nickvergessen adjusted the schema for you, now only needs docs

File: nextcloudappstore/api/v1/release/info.xsd (hunk 1 above)
Comment (nickvergessen) [reply nextcloudappstore/api/v1/release/info.xsd]: done

**Pull request merged**